### ISSUE TITLE
No confirmation on testrpc with simple dapplescript

### ISSUE THREAD

**Issue by raineorshine** [nexusdev/dapple#248]
I created this simple dapplescript for a trivial Box contract:
```
var box = new Box.value(1000000)("Box")
box.Set.value(100000)(42)
```

My `~/.dapplerc`:
```
environments:
  default:
    ethereum: internal
  evm: default
  live:
    ethereum:
      host: localhost
      port: '8545'
      account: '0x825c74b1fbec73911379b30e1b0ba89ff40e662a'
    ipfs:
      host: localhost
      port: '5001'
```

When I try to run the dapplescript against my testrpc, it only makes it as far as deploying the Box contract (though no confirmation):

```
$ dapple run foo.dapplescript -e live
/bin/sh: solc: command not found
No local solc found. Failing over to JS compiler...
simulated successfully
Deploying to environment "live"...
Connected to RPC client, block height is 0
DEPLOYED: Box = 0x47bc69e60fc5651f5efd35ba0fd60de8e6a19b1e
GAS COST: 90101 for "new Box"
[hangs]
```

My testrpc just shows `eth_getFilterChanges` over and over:
```
Listening on localhost:8545
eth_getBlockByNumber
eth_getBlockByNumber
eth_blockNumber
eth_newBlockFilter
eth_sendTransaction
eth_getFilterChanges

  Contract created: 0x47bc69e60fc5651f5efd35ba0fd60de8e6a19b1e
  Gas usage: 90101
  Block Number: 0x01

eth_getFilterChanges
eth_getFilterChanges
eth_getTransactionReceipt
eth_getCode
eth_uninstallFilter
eth_newBlockFilter
eth_getFilterChanges
eth_getFilterChanges
eth_getFilterChanges
eth_getFilterChanges
eth_getFilterChanges
eth_getFilterChanges
eth_getFilterChanges
eth_getFilterChanges
...
```

When I run the dapplescript on the `internal` environment, it completes successfully:

```
$ dapple run foo.dapplescript --report
/bin/sh: solc: command not found
No local solc found. Failing over to JS compiler...
Deploying to environment "default"...
Using internal EVM...
Connected to RPC client, block height is 0
DEPLOYED: Box = 0x01acf0e8367a20b5029716b9d7d1968d824733df
GAS COST: 90101 for "new Box"
CONFIRMED deploy after 1 blocks!
CALLED: Box("0x01acf0e8367a20b5029716b9d7d1968d824733df").Set(42)
GAS COST: 41624 for "call Box.Set(42)"

.
  Successfully deployed!
```

Why doesn't it complete on testrpc? Thank you for your help.

**Comment by raineorshine** [comment]
```
OSX 10.11.4
Dapple v0.7.1
EthereumJS TestRPC v2.0.5
```

**Comment by mhhf** [comment]
Thank you for the detailed report!
TLDR: add `confirmationBlocks: 0` to your live environment:
```
environments:
    live:
        confirmationBlocks: 0
...
```

This is an issue with testrpc we know about, this shows me that we need a better documentation to prevent this from happening. DappleScript has a default confirmation time of 1 Block: It waits 1 block to verify if your transaction got included and no fork occurred. TestRPC only mines blocks, if something new happens -- that produces a deadlock where dapple waits for a new block to confirm your previous transaction and TestRPC for a new transaction to mine a new block. Setting `confirmationBlocks: 0` turns the block confirmation off.

**Comment by raineorshine** [comment]
Awesome. Glad it is a known issue.

Can dapple detect that it is connecting to testrpc instead of, for example, geth? If so, maybe it can automatically set `confirmationBlock` to 0.

**Comment by mhhf** [comment]
It could. Will address this when usability gets priority

**Comment by mhhf** [comment]
Added to testrpc to FAQ in readme section, will close this and address usability in a separate issue
